src/test/kotlin/com/atlassian/performance/tools/report/RootWorkspaceTest.kt

```diff
@@ -0,0 +1,30 @@
+package com.atlassian.performance.tools.report
+
+import com.atlassian.performance.tools.workspace.api.RootWorkspace
+import org.assertj.core.util.Files
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.Test
+
+class RootWorkspaceTest {
+
+    @Test
+    fun shouldCreatingNewInstanceHaveNoSideEffect() {
+        // given
+        val tempDir = Files.newTemporaryFolder()
+        // when
+        RootWorkspace(parent = tempDir.toPath())
+        // then
+        assertThat(tempDir).isEmptyDirectory()
+    }
+
+    @Test
+    fun shouldCreateDirWhenGetCurrentTask() {
+        // given
+        val tempDir = Files.newTemporaryFolder()
+        // when
+        RootWorkspace(parent = tempDir.toPath()).currentTask
+        // then
+        assertThat(tempDir).isNotEmptyDirectory()
```

Comment:
What does it contain in this case? Just the `RootWorkspace.directory` or `RootWorkspace.currentTask.directory`?
If the problem is "sometimes we'll have empty directories", then this still happens here, right?

Reply:
`RootWorkspace.directory` until `TaskWorkspace.directory` is read

Reply:
I've changed the test to reflect better what's actually tested